### ISSUE TITLE
[FEATURE]: Player section. Ability to delete player individually.

### ISSUE THREAD

**Issue by Legacy-TacticalGamingInteractive** [citizenfx/txAdmin#751]
### Scope

Web

### Feature Description

Ability to delete an individual user via the Player section as at the moment I believe there is only ban as an option but that is not always applicable. Pruning individual users can be useful as well in some situations.

**Comment by MasterVargen** [comment]
Hi 
There is a kick button or how do you mean?

**Comment by Legacy-TacticalGamingInteractive** [comment]
I mean as in remove just one individual from the database. Rather than a total wipe.

**Comment by tabarra** [comment]
@Legacy-TacticalGamingInteractive Can you please elaborate in which situations removing a player could be useful?

**Comment by Legacy-TacticalGamingInteractive** [comment]
> @Legacy-TacticalGamingInteractive Can you please elaborate in which situations removing a player could be useful?

Case 1) We've had a handful of times such as today which reminded me where someone for some reason has a second Discord linked to their identifiers. This person swears they don't own that account nor know who it is. Maybe someone logged in on their PC once or they on someone elses and they do not recall. Either way,

I was hoping to just remove them from the database individually so they could reconnect and it can grab fresh identifiers from them perhaps? If thats how I could go about it.

FiveM/tx thinks two people are logging in at the same time even though hes not logged into that other account he has no access to. Its showing the wrong Discord for him in game than the one hes logged in on as well. Due to this, his permissions are broken as well as ours are tied to Discord Roles.

Case 2) Its also happened where we had to ban someone who shared a computer with someone else in the past, but no longer does. Because of this reason, the other person who shared the computer who isn't banned can't login either anymore since the identifiers are linked from the time they shared a computer. The other person has their own computer now but is still unfortunately banned because of it. So I cannot pull fresh identifiers for them and unban just them.

(I suppose another alternative to this, though tricky no doubt, would be removing individual identifiers lol. But that sounds too hard to implement and likely not worth the time to do so?)

**Comment by Mycroft-Studios** [comment]
Case 1: That literally isn't how the identifiers work.
for it to be in that list, they had to specifically authorise thst discord account, and knowingly join your server with that account linked.

> FiveM thinks two people are logging in at the same time on the same.

If it thinks that, that is whats happening. and the user is lying to you. ( the only time this wouldn't be the case , is if the client crashes, in which case the server takes a few minutes to recognise the fact the are not there)

Case 2: While I can see this actually happening, I *do* personally , feel this is would allow for abuse. Identifiers are not 100% secure , they *can* be spoofed. 99% , if not 100% of the times we have seen this exact scenario, It's the EXACT same person , who tried to spoof the identifiers , but because txadmin stores past ones aswell, it doesn't work, so they goto the server owner and lie to get unbanned. ( I want to point out we have seen this atleast 50 times , and this has always been the outcome).

So while , I can see why this might be needed, in the like 1% of times it might not be. I personally don't think it should be added. that is just my opinion though

**Comment by Legacy-TacticalGamingInteractive** [comment]
The first use case. Well the latest person in question has no reason to use the other account even if they wanted to and had access as their main account has lots of roles and permissions. They aren't in trouble and have played with us for a year. They aren't banned they are just not able to get their permissions in game now because of it.

As for case 2... I can see if someones trying to spoof shit why that would be an issue. But the person who is the person that is not banned in this case is actually a Staff member who shared an old PC with the person who did get banned. They are in fact two totally different people. But he cannot be unbanned because the other guy is. So I mean. Its a bit of a pickle.

Having the option is not a bad thing. If you don't want to use it then don't.

**Comment by Mycroft-Studios** [comment]
> But the person who is the person that is not banned in this case is actually a Staff member who shared an old PC with the person who did get banned

this is due to hardware identitifers and why the bans are now super powerful. Thanks to this, someone who got banned , could literally buy an entire new computer , and still be banned.

Imo, it would be useful have a way to Clear the stored , past identifiers of a player, in this situation. Rather than deleting the entire player in the data.

**Comment by Legacy-TacticalGamingInteractive** [comment]
Yeah either way some way to either remove them entirely so it picks up new identifiers only from their new PC in specific use cases. Or as you say what I mentioned if possible which is to clear identifiers for someone so it can pick up proper ones. Either or.

**Comment by tabarra** [comment]
Update on this request: I'm considering adding something among those lines. Though, would probably be just an option to wipe player IDs instead of deleting a players.

**Comment by Legacy-TacticalGamingInteractive** [comment]
> Update on this request: I'm considering adding something among those lines. Though, would probably be just an option to wipe player IDs instead of deleting a players.

Sounds good. Whatever is feasible really.

**Comment by Legacy-TacticalGamingInteractive** [comment]
Is this still something thats possible. Ran into the issue again with someone logging in on their friends PC.

**Comment by adhershmnair** [comment]
+1
